src/waveform/renderers/glslwaveformrenderersignal.cpp

```diff
@@ -219,14 +219,29 @@ void GLSLWaveformRendererSignal::onSetup(const QDomNode& node) {
 }
 
 void GLSLWaveformRendererSignal::onSetTrack() {
-    m_loadedWaveform = 0;
-    loadTexture();
+    slotWaveformUpdated();
+
+    TrackPointer pTrack = m_waveformRenderer->getTrackInfo();
+    if (!pTrack) {
+        return;
+    }
+
+    // When the track's waveform has been changed (or cleared), it is necessary
+    // to update (or delete) the texture containing the waveform which was
+    // uploaded to GPU. Otherwise, previous waveform will be shown.
+    connect(pTrack.get(), SIGNAL(waveformUpdated()),
```

Comment:
What about disconnecting the signal? I think this means clearing a track also reloads the texture for all decks it was loaded in (even if a new track has replaced it).